tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/api/validation/PostValidator.java

```diff
@@ -306,8 +307,16 @@ private Collection<? extends FieldError> checkFunding(PostDTO postDto) {
             && !postFundingDto.getEndDate().isAfter(postFundingDto.getStartDate())) {
           fieldErrors.add(new FieldError(POST_DTO_NAME, "fundings",
               "Post funding end date must not be equal to or before start date"));
+        } else if (postFundingDto.getFundingType() == null) {
+          fieldErrors.add(
+              new FieldError(POST_DTO_NAME, "fundings",
+              "Post Funding must have a funding type"));
         }
       }
+    } else if (postDto.getStatus() == Status.CURRENT) { //No Post Funding but CURRENT
+      fieldErrors.add(
+          new FieldError(POST_DTO_NAME, "fundings",
+              "Current Posts Must have at least 1 Post Funding"));
```

Comment:
PostDto.status is not required and is populated from post fundings, so I think backend should not rely on the status field from the client.

Reply:
Do you have any advice on a good approach for this? It feels odd to be fetching from the DB at this point but I guess that's the only option given that the field is a backend-only managed one?

Reply:
I think the `validate` method is only used by createPost & updatePost methods in PostResource.
And the client should send all the post fundings attached to the post to the backend. So if you want to check if there's one postFunding, you can just check the dto directly.